docs/terraform/common-troubleshooting.mdx

```diff
@@ -0,0 +1,106 @@
+---
+title: "Common Troubleshooting & Recipes"
+description: "Learn about overcoming API design incompatibilities when generating Terraform providers with Speakeasy."
+---
+
+# Common Troubleshooting & Recipes
+
+## Overview
+
+When generating Terraform providers from OpenAPI specifications, you might
+encounter cases where your API’s structure doesn’t naturally fit Terraform’s
+resource-oriented approach. These differences typically occur because APIs
+aren’t always designed with Terraform’s infrastructure management style in
+mind.
+
+Speakeasy’s generator identifies these design differences and offers extensions
+and customization options to address them. This enables you to produce
+effective Terraform providers, even when your API doesn’t initially match
+Terraform’s requirements, simplifying what might otherwise be challenging
+configuration tasks.
+
+## Fixing Common API Issues
+
+### Impedance Mismatch Errors
+
+An impedance mismatch error happens when Speakeasy’s generator finds properties
+with different data types that need to be combined into one. This error means
+the request and response data don’t match up correctly, which can cause
+problems in the Terraform provider that’s being created.
+
+#### Common example
+
+A typical impedance mismatch scenario occurs in the following situation:
+
+- A request takes UUID string for task assignment:
+
+```json request.json
+{
+  "taskId": "12345",
+  "assignee": "user-uuid-1"
+}
+```
+
+- The response returns a full user object for that assignee:
+
+```json response.json
+{
+  "taskId": "12345",
+  "assignee": {
+    "id": "user-uuid-1",
+    "name": "Alice Johnson"
+  }
+} 
+```
+
+When the generator attempts to merge these properties, it detects that it
+cannot reconcile the different data types (string vs object) and yields an
+impedance mismatch error.
+
+#### How to Fix the Problem
+
+Speakeasy includes built-in extensions to help fix impedance mismatch errors.
+You can use these solutions to adjust either the request or the response data:
+
+**Option 1: Ignore the mismatched property**
```

Comment:
❤️ for calling out multiple options! I think we should suggest name override first. It might also be good to mention the tradeoffs for both options:

* Name override: Additional data is still available to Terraform consumers, but potential loss of drift detection depending on operation.
* Ignore: Same as above but data may no longer be available to Terraform consumers.